scim2-sdk-common/src/test/java/com/unboundid/scim2/common/FilterEvaluatorTestCase.java

```diff
@@ -399,6 +398,25 @@ public void testCombiningFilterEquals() throws Exception
     );
     assertThat(orFilter.equals(orFilterDifferentOrder)).isTrue();
     assertThat(orFilterDifferentOrder.equals(orFilter)).isTrue();
+
+    // Test nested AND/OR filters.
+    Filter filter1 = Filter.fromString(
```

Comment:
I incorporated this test case since it exposes a regression that occurred with an earlier version of this update. It seemed useful to include that in our own unit tests.